src/useRouteQueryParam/formats/RouteParam.ts

```diff
@@ -58,7 +58,7 @@ export abstract class RouteParam<T> {
   }
 
   public set(query: LocationQuery, value: T | T[] | undefined): void {
-    if (value === undefined) {
+    if (value === undefined || value === null) {
```

Comment:
this isn't strictly part of the fix but we use `Nullable*` for lots of our route classes which meant that they were never getting removed - happy to revert this

Reply:
personally I think we lose functionality here. But maybe this isn't how its working? Hard to remember. 
- 1 - `?foo=1`
- null - `?foo`
- undefined - `?`

Reply:
Hm, maybe it is working like that, that wasn't intuitive to me; I'm gonna revert since it seems like this is intended behavior at the least

Reply:
I believe that reflects how vue-router works as well.